iPetCare/Application/Services/UserService.cs

```diff
@@ -1,146 +1,185 @@
-                return new ServiceResponse<RegisterDtoResponse>(HttpStatusCode.BadRequest, errors);
-            }
-            else
-                return new ServiceResponse<RegisterDtoResponse>(HttpStatusCode.BadRequest, "Podana rola nie istnieje");
+                Id = userToRegister.Id,
+                FirstName = userToRegister.FirstName,
+                LastName = userToRegister.LastName,
+                Email = userToRegister.Email,
+                UserName = userToRegister.UserName,
+                Role = userToRegister.Role,
+                Token = _jwtGenerator.CreateToken(userToRegister)
+            };
+            return new ServiceResponse<RegisterDtoResponse>(HttpStatusCode.OK, responseDto);
         }
 
-        public async Task<ServiceResponse<GetAllDtoResponse>> GetAllAsync()
+        private async Task<ServiceResponse<RegisterDtoResponse>> ValidateRegisterRequestAsync(RegisterDtoRequest dto)
         {
-            var currentUserName = _userAccessor.GetCurrentUsername();
+            if (await Context.Users.Where(x => x.Email == dto.Email).AnyAsync())
+                return new ServiceResponse<RegisterDtoResponse>(HttpStatusCode.BadRequest, "Email jest zajęty");
 
-            if (currentUserName == null)
-                return new ServiceResponse<GetAllDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+            if (await Context.Users.Where(x => x.UserName == dto.UserName).AnyAsync())
+                return new ServiceResponse<RegisterDtoResponse>(HttpStatusCode.BadRequest, "Nick jest zajęty");
+            return new ServiceResponse<RegisterDtoResponse>(HttpStatusCode.OK);
+        }
 
-            var currentUser = await _userManager.FindByNameAsync(currentUserName);
-            if (currentUser == null)
-                return new ServiceResponse<GetAllDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
-            if(currentUser.Role != Role.Administrator)
-                return new ServiceResponse<GetAllDtoResponse>(HttpStatusCode.Forbidden, "Brak uprawnień");
+        private async Task<ServiceResponse<RegisterDtoResponse>> CreateUserAsync(RegisterDtoRequest dto, ApplicationUser userToRegister)
+        {
+            var result = await UserManager.CreateAsync(userToRegister, dto.Password);
+            if (result.Succeeded) return new ServiceResponse<RegisterDtoResponse>(HttpStatusCode.OK);
 
-            var users = await _context.Users.ToListAsync();
+            var errors = string.Join("\n", result.Errors.Select(x => x.Description));
+            return new ServiceResponse<RegisterDtoResponse>(HttpStatusCode.BadRequest, errors);
+        }
 
-            var dto = new GetAllDtoResponse();
-            dto.Users = _mapper.Map<List<UserGetAllDtoResponse>>(users);
+        private async Task<ServiceResponse<RegisterDtoResponse>> CreateVetAsync(ApplicationUser userToRegister, RegisterDtoResponse responseDto)
+        {
+            var vet = new Vet
```

Comment:
vet powinien jeszcze mieć Institution dodawane (chyba, że uznajemy, że jest opcjonalne)

Reply:
jest opcjonalne - to rejestracja ogólnego appusera, reszte uzupełnią sobie po zalogowaniu :)